### DISCUSSION TITLE
Goodbye

### DISCUSSION THREAD

**Discussion by ghost** [modula3/cm3#1154] [General]
After many months trial of CM3 I found M3 is unsuitable for my needs. I will go back to C++ and I will have to learn a lot to update my skills to match modern C++ because my knowledge of C++ is stuck at C+98. Thank you so much for being patient with me and helped me. Sorry but I must go, I found myself being unproductive with CM3. I tried but I can't do anything useful with it.

**Comment by demoitem** [comment]
Oh, what a shame. I was quite enjoying your frequent reappearances after those theatrical exits. Rather like Banquo's ghost.
I'm sure you will find the people in the C++ groups welcoming. I've heard they are a cheerful lot.
Certainly, the designers of that language possess a sense of humour. Just take a look at their 2000 page standard. 
But, there are a multitude of languages out there to choose from - from Ballerina to F#. (Actually, I read that the management of Sibelius were planning a rewrite, and F# had struck a chord - just a minor one. However, a few key engineers on the staff thought that C# would be a lower bar to success. The tone became quite ugly in the end and the project fell flat).
So, good luck in the land of undefined behaviour. Safety-critical guys love it.
Pop back sometime and let us know how you got on.

**Reply by dmuysers** [reply]
Hahaha! The best posting for years!

**Reply by ghost** [reply]
I do not regret coming back to C++! As a normal developer like me you will find yourself most of the time not writing your own shining things from scratch but calling people's code (in the form of external libraries). This is a nightmare on M3 but a non-existent problem on C++. Only if I'm a professional developer of big corporations like Google and Cloudflare then I should care about the language I'm gonna to use is "safe" or not as I could have to write an entire new webserver from scratch using said language. This is not the case here!

**Comment by RodneyBates** [comment]
Touche'!

On 2/13/23 03:27, peter mckinna wrote:
>
> Oh, what a shame. I was quite enjoying your frequent reappearances after those theatrical exits. Rather like Banquo's ghost.
> I'm sure you will find the people in the C++ groups welcoming. I've heard they are a cheerful lot.
> Certainly, the designers of that language possess a sense of humour. Just take a look at their 2000 page standard.
>
I didn't realize it had reached 2000.  Is that including the libraries?
Last I knew, it was a mere 600, but that was only the language proper.
